dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -405,4 +412,41 @@ private boolean isClassAccessibleFrom(JavaClass javaClass, String sourceClassNam
       return false;
     }
   }
+
+  /**
+   * Returns an {@code Optional} describing the symbol problem in the parent classes or interfaces
+   * for {@code baseClassName}, if any of them are missing; otherwise an empty {@code Optional}.
+   */
+  private Optional<SymbolProblem> findParentSymbolProblem(String baseClassName) {
+    Queue<String> queue = new ArrayDeque<>();
+    queue.add(baseClassName);
+    while (!queue.isEmpty()) {
+      String className = queue.remove();
+      if (Object.class.getName().equals(className)) {
+        continue; // java.lang.Object is the root of the inheritance tree
+      }
+      JavaClass baseClass;
+      try {
+        baseClass = classDumper.loadJavaClass(className);
+        queue.add(baseClass.getSuperclassName());
+      } catch (ClassNotFoundException ex) {
+        // baseClass is not updated yet
+        SymbolProblem problem =
+            new SymbolProblem(new ClassSymbol(className), ErrorType.SYMBOL_NOT_FOUND, null);
+        return Optional.of(problem);
+      }
+
+      for (String interfaceName : baseClass.getInterfaceNames()) {
```

Comment:
This and the following should also be pushed into the try block.

Reply:
No, I tried that but it would not work because interfaceName is used in the catch clause.

Reply:
I meant this should be pu8shed into the try block above, not below. You can nest the try blocks.

Reply:
I was able to remove second try block by introducing `potentiallyMissingTypeName` variable ([link](https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/935/files#diff-e275d9cce5c577df64bfd6fe55a5cdfaR434)). Does that look good?